### PR TITLE
NIP-60 + NIP-61 Implementation

### PR DESCRIPTION
**Pull request by lindsaymoralesb**
## Issue
Partially solves #168 

## Description
- Hooks added for NIPs 60 (wallet event, token event, spending history event) and 61 (nut zap event).
- Hooks integrated to keep wallet state in nostr.
- Added share ecash gift link.

## Demo
- Ecash share

https://github.com/user-attachments/assets/77427c92-4865-4350-8a1e-3030bbace92f

- Ecash gift link receive:

https://github.com/user-attachments/assets/ad9bda43-ffc2-4833-8159-411d3e92335d

### CODE REVIEW

File: apps/mobile/src/modules/CashuWallet/index.tsx [hunk 1]
```diff
@@ -116,6 +125,22 @@ export const CashuView = () => {
     setActiveUnitStorage(data.units[0]);
     setMintsStorage([data]);
     setAddingMint(false);
+
+    const privKey = getRandomBytes(32);
```
Comment (MSghais) [inline apps/mobile/src/modules/CashuWallet/index.tsx:129]: Why create another key here?

File: apps/mobile/src/modules/CashuWallet/components/Send/index.tsx [hunk 2]
```diff
@@ -93,10 +93,12 @@ export const Send: React.FC<SendProps> = ({onClose}) => {
     return ecash;
   };
 
-  const handleCopy = async (type: 'ecash') => {
+  const handleCopy = async (type: 'ecash' | 'link') => {
     if (!generatedEcash) return;
     if (type == 'ecash') {
       await Clipboard.setStringAsync(generatedEcash);
+    } else if (type === 'link') {
+      await Clipboard.setStringAsync(`/receive/ecash/${generatedEcash}`);
```
Comment (MSghais) [inline apps/mobile/src/modules/CashuWallet/components/Send/index.tsx:101]: Looks the Screen is not created and not in the Router right now?

**Review by MSghais — CHANGES_REQUESTED** [review]

File: apps/mobile/src/modules/CashuWallet/index.tsx (hunk 1 above)
Comment (lindsaymoralesb) [reply apps/mobile/src/modules/CashuWallet/index.tsx:129]: From nostr docs ([NIP-60](https://github.com/nostr-protocol/nips/blob/master/60.md)):

Screenshot 2024-11-26 at 7 04 42 AM

**Review by MSghais — APPROVED** [review]
LFG good game! Yoollo merged and let's see and check more deeply after and iterate

**Pull request merged**